client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -471,6 +486,34 @@ export class PlanFeatures2023Grid extends Component<
 			} );
 	}
 
+	/**
+	 * Similar to `renderTable` above.
+	 */
+	renderCurrentPlan( planPropertiesObj: PlanProperties[] ) {
```

Comment:
I'd prefer if we rename this to something like `renderPlanHighlight`. 

Although - is it too much work to turn this into a function component instead? That's the type of refactoring that will happen soon in this file (probably by me), so might as well do it here if not too heavy (especially if going with a [non-table view](https://github.com/Automattic/wp-calypso/pull/78845/files#r1252475292))

Reply:
> I'd prefer if we rename this to something like renderPlanHighlight.

Done via https://github.com/Automattic/wp-calypso/pull/78845/commits/d2e44b92c9b93b95ea6781271c94ec9d51b657a0!

Reply:
> Although - is it too much work to turn this into a function component instead? That's the type of refactoring that will happen soon in this file (probably by me), so might as well do it here if not too heavy (especially if going with a [non-table view](https://github.com/Automattic/wp-calypso/pull/78845/files#r1252475292))

I started working on this task by writing a fresh functional component but realized that I was duplicating a lot of existing code and styles. This is a class component, and much of the rendering is done by `renderXYZ` style functions which pick their required props from `this.props`. To reuse these functions, each of these `renderXYZ` functions would have to be converted to functional components. This would be a lengthy task, and I feel it would be best if tackled separately. Definitely something to do down the line, though!

Reply:
Maybe we can do a tiny bit here to help save a few clicks from whoever does that refactor next. Is it possible to create the function component in a separate file and pass these other components through as props or children to render them?

Reply:
I believe we should fix the scope and maintain the emphasis on highlighting the current plan in this PR. Introducing refactors may complicate matters for reviewers and testers. Perhaps it would be more convenient to start the refactoring change afresh while looking at the complete picture instead of attempting to integrate a half-baked change into this PR. The new function, "renderPlanHighlight," is small enough to be refactored with ease and utilizes existing functions without introducing any novel concepts.

Reply:
> may complicate matters for reviewers and testers

I guess as the reviewer in this case, this would be my suggestion. also in context with everything else happening for [wider refactors](https://github.com/Automattic/wp-calypso/labels/Plans%3A%20NPM%20essential).

Reply:
I'm happy to pick this up in a separate PR, as this particular refactor (class function -> functional component) is not contributing to the goal of the current PR. What do you think about moving this PR forward and tackling this separately?